src/App.js

```diff
@@ -289,34 +292,91 @@ export default class App extends React.Component {
     <>
-    <Route path='/' exact render={() => <HomeNavBar loggedIn={this.state.auth.loggedIn} onLogout={this.logout} />}/>
-    <Route path='/dashboard' exact render={() => <HomeNavBar loggedIn={this.state.auth.loggedIn} onLogout={this.logout} />}/>
-    <Route path='/feeling_tracker' exact render={() => <HomeNavBar loggedIn={this.state.auth.loggedIn} onLogout={this.logout} />}/>
-    <Route path='/todos' exact render={() => <HomeNavBar loggedIn={this.state.auth.loggedIn} onLogout={this.logout} />}/>
-    <Route path='/journal' eaxct render={() =><HomeNavBar loggedIn={this.state.auth.loggedIn} onLogout={this.logout} />}/>
+      <Route
+        path="/"
+        exact
+        render={() => (
+          <HomeNavBar
+            loggedIn={this.state.auth.loggedIn}
+            toggleClassOnNavOpen={this.toggleClassOnNavOpen}
+            showNav={this.state.showNav}
+          />
+        )}
+      />
+      <Route
+        path="/dashboard"
+        exact
+        render={() => (
+          <HomeNavBar
+            loggedIn={this.state.auth.loggedIn}
+            toggleClassOnNavOpen={this.toggleClassOnNavOpen}
+            showNav={this.state.showNav}
+          />
+        )}
+      />
+      <Route
+        path="/feeling_tracker"
+        exact
+        render={() => (
+          <HomeNavBar
+            loggedIn={this.state.auth.loggedIn}
+            toggleClassOnNavOpen={this.toggleClassOnNavOpen}
+            showNav={this.state.showNav}
+          />
+        )}
+      />
+      <Route
+        path="/todos"
+        exact
+        render={() => (
+          <HomeNavBar
+            loggedIn={this.state.auth.loggedIn}
+            toggleClassOnNavOpen={this.toggleClassOnNavOpen}
+            showNav={this.state.showNav}
+          />
+        )}
+      />
+      <Route
+        path="/journal"
+        eaxct
+        render={() => (
+          <HomeNavBar
+            loggedIn={this.state.auth.loggedIn}
+            toggleClassOnNavOpen={this.toggleClassOnNavOpen}
+            showNav={this.state.showNav}
+          />
+        )}
+      />
     </>
-  )
+  );
 
   renderFooter = () => (
     <>
-    <Route path='/' exact render={() => <Footer />}/>
-    <Route path='/dashboard' exact render={() => <Footer />}/>
-    <Route path='/feeling_tracker' exact render={() => <Footer />}/>
-    <Route path='/todos' exact render={() => <Footer />}/>
-    <Route path='/journal' eaxct render={() =><Footer />}/>
+      <Route path="/" exact render={() => <Footer />} />
+      <Route path="/dashboard" exact render={() => <Footer />} />
+      <Route path="/feeling_tracker" exact render={() => <Footer />} />
+      <Route path="/todos" exact render={() => <Footer />} />
+      <Route path="/journal" eaxct render={() => <Footer />} />
     </>
-  )
+  );
+
+  //toggle boolean state value to to control Nav overlay render 
+  //this function and 'showNav' are passed to HomeNavBar
+  toggleClassOnNavOpen = () => {
+    this.setState(prevState => ({
```

Comment:
My first idea would be to put a `debugger` statement between lines 367-368, and then ask:
1. Do I hit this function when I expect to?
2. When I hit it, what is the previous state & the current state?